### ISSUE TITLE
"Sliding" widgets in traffic module

### ISSUE THREAD

**Issue by somospocos** [tobi-wan-kenobi/bumblebee-status#567]
### Bug Report
"Siding"  widgets in traffic module

#### Summary
Affected module: traffic

#### Description:
I am opening this issue to keep it separate from the more general previous one.

The goal of this issue is to find a consensus for fixing the sliding trouble for this particular module.

I did some manual tests and I so far I found two independent reasons why it's happening.

1. My assumption from the other issue https://github.com/tobi-wan-kenobi/bumblebee-status/blob/a7d22edcdaba208f4bf4115cc012f53ffd90cbbc/bumblebee/util.py#L64 proved to be true this time. Changing this line to

        for unit in ["  ", "Ki", "Mi", "Gi"]
    improves the situation, but leaves a gap between the numeric value and the units when   it's just B/s. Personally, I'm fine with this side effect. My question about this: is it possible to achieve the same effect without modifying said line of code with just a custom -p traffic.format string? So far I don't know, due to still not knowing much about format strings. If it's not possible to do it via format string, shall we change the said line of code? I'm strongly in favor of it.

2. Second reason is that widgets slide because of variable length of the integer numeric value  returned by util.bytefmt(). The default value of format string is unfortunate, a saner one to prevent sliding would be `-p traffic.format="{:7.2f}"`. As I understand, this means "the numeric value before the units is 7 chars, two after the dot + the dot + other 4, which is the length of '1024'".

TLDR: I was able to fix this particular sliding widget problem for traffic module by padding the empty list item for bytes per second and using a better format string `-p traffic.format="{:7.2f}"`

**Comment by tobi-wan-kenobi** [comment]
Actually, looking at the code, I am somewhat surprised that the "minwidth" calculation is commented out... That looks like a bug.

A user *should* be able to specify the minimum width of a module/widget using `theme.minwidth`, not sure why I took that out. Possibly due to issues with the module vs. widget concept? (It's easy to specify parameters for modules, but hard for widgets, if the module is multi-widget).

Will investigate over the weekend, but for the short-term, I'd say using a saner format string sounds like the way to go.

**Comment by somospocos** [comment]
Which location in the code are you referring to? I did a `rg minwidth` in the root of the repository and I don't see any code related to minwidth that is commented.

**Comment by somospocos** [comment]
git blame shows the line above is related to changes made by me, I will investigate the history more, maybe I remember.

**Comment by tobi-wan-kenobi** [comment]
It's in output.py, setting the min_width attribute. The first git blame showed a change where the prefix/postfix were made members, but the min_width line was commented then, as well...

**Comment by somospocos** [comment]
revision 4a7a2645

the diff from that revision is 

    -#            "min_width": width + "A"*(len(prefix) + len(suffix)) if width else None,
    +#            "min_width": width + "A"*(len(self._prefix) + len(self._suffix)) if width else None,

Now I remember - it was already commented, but since I refactored and made
prefix and suffix instance variables, because I was planning to extract the
drawing logic into its own class and use these in more than one method of that
class. I thought that since it was commented, rather than removed, it might be
needed later, so I went and adjusted that commented line as well, just in case
if it will need to be uncommented in the future.

The commit before that is 9d8c1f91

The comment happened before that, `git blame 9d8c1f9` shows:


    7d55bdce (Tobias Witek    2018-11-25 17:50:05 +0100 404)             "min_width": None,
    47eeac00 (Tobias Witek    2019-01-26 19:40:08 +0100 405) #            "min_width": width + "A"*(len(prefix) + len(suffix)) if width

The commit `47eeac00` happened long before I started to use bumblebee-status
and it's a revert commit.

**Comment by somospocos** [comment]
It seems to be this commit that made the change 7d55bdce

**Comment by tobi-wan-kenobi** [comment]
Well, it *did* say it was experimental in the commit :P

Joking aside, I think I found the issue - it's in the traffic module. Let me do the commit and see whether it improves your situation.

**Comment by somospocos** [comment]
After doing some testing with #569 and #570 , it looks good.

**Comment by tobi-wan-kenobi** [comment]
Just as an aside: From my latest tests in development (the CPU module works again - yay), seems like the original restriction (click events don't register in the "padded" area) are gone - so I will stay with `min_width` as i3 protocol field there.
